diff --git a/layer1/PyMOLObject.cpp b/layer1/PyMOLObject.cpp
index 1e56c0d5..cd9e5998 100644
--- a/layer1/PyMOLObject.cpp
+++ b/layer1/PyMOLObject.cpp
@@ -14,6 +14,8 @@ I* Additional authors of this source file include:
 -*
 Z* -------------------------------------------------------------------
 */
+#include <algorithm>
+
 #include"os_python.h"
 
 #include"os_predef.h"
diff --git a/layer1/ScrollBar.cpp b/layer1/ScrollBar.cpp
index 6c6d7f3f..066bd683 100644
--- a/layer1/ScrollBar.cpp
+++ b/layer1/ScrollBar.cpp
@@ -14,6 +14,8 @@ I* Additional authors of this source file include:
 -*
 Z* -------------------------------------------------------------------
 */
+#include <algorithm>
+
 #include"os_gl.h"
 #include"Base.h"
 #include"Block.h"
